File: LegoCatalog.DotNet.API/LegoCatalog.Data/Part.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LegoCatalog.Data 
{
    [Table("PartsCatalog")]
    public class Part 
    {
        public int PartId { get; set; }

        [Key]
        public string ItemId { get; set; }
        public string ItemName { get; set; }
        
        public string ItemTypeId { get; set; }
        [ForeignKey("ItemTypeId")]
        public ItemType ItemType { get; set; }

        public int CategoryId { get; set; }
        [ForeignKey("CategoryId")]
        public Category Category { get; set; }

        public List<PartColor> PartColors { get; set; }
        
        public decimal? ItemWeight { get; set; }
        public decimal? ItemDimensionX { get; set; }
        public decimal? ItemDimensionY { get; set; }
        public decimal? ItemDimensionZ { get; set; }
        public string ImageLink { get; set; }
        public string IconLink { get; set; }
        public int Quantity { get; set; }

        [NotMapped]
        public int ColorQuantity { get; set; }
    }
}

File: LegoCatalog.ng.client/LegoCatalog-ng-client/src/app/part.ts
import { PartColor } from './partColor';

export class Part {
  partId: number;
  itemId: string;
  itemName: string;
  itemTypeId: string;
  itemTypeName: string;
  categoryId: number | null;
  categoryName: string | null;
  itemWeight: number | null;
  itemDimensionX: number | null;
  itemDimensionY: number | null;
  itemDimensionZ: number | null;
  imageLink: string | null;
  iconLink: string | null;
  quantity: number | null;
  colorCount: number | null;
  partColors: PartColor[] | null;
}
